### ISSUE TITLE
[Feat]Name resolution for node_id in SMF

### ISSUE THREAD

**Issue by sh-sekigawa** [free5gc/free5gc#111]
### What's the version are you using
v3.0.3

### Is your feature request related to a problem? Please describe.
I'm trying to run Free5GC on Kubernetes.
The container images are based on https://github.com/free5gc/free5gc-compose.
I'd like to use hostname to configure node_id in smfcfg.conf, because the IP addresses for Pod is assigned dynamically.
Currently, SMF doesn't resolve hostname and treats the string as an IP address.

SMF logs when docker-compose up are as below.
```
$ sudo docker logs smf
smf
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/src/smf/smf.go:35 main.main() SMF version:  2020-03-xx-xx                   SMF=app
CommonConfig file: ../config/free5GC.conf
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/lib/nas/logger/logger.go:60 free5gc/lib/nas/logger.SetLogLevel() set log level : info                          NAS=message
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/lib/nas/logger/logger.go:65 free5gc/lib/nas/logger.SetReportCaller() set report call : true                        NAS=message
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/lib/fsm/logger/logger.go:56 free5gc/lib/fsm/logger.SetLogLevel() set log level : info                          fsm=fsm
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/lib/fsm/logger/logger.go:61 free5gc/lib/fsm/logger.SetReportCaller() set report call : true                        fsm=fsm
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/lib/ngap/logger/logger.go:56 free5gc/lib/ngap/logger.SetLogLevel() set log level : info                          NGAP=NGAP
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/lib/ngap/logger/logger.go:61 free5gc/lib/ngap/logger.SetReportCaller() set report call : true                        NGAP=NGAP
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/lib/openapi/Namf_Communication/logger/logger.go:57 free5gc/lib/openapi/Namf_Communication/logger.SetLogLevel() set log level : info                          Namf=Comm
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/lib/openapi/Namf_Communication/logger/logger.go:62 free5gc/lib/openapi/Namf_Communication/logger.SetReportCaller() set report call : true                        Namf=Comm
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/lib/openapi/Namf_EventExposure/logger/logger.go:57 free5gc/lib/openapi/Namf_EventExposure/logger.SetLogLevel() set log level : info                          Namf=Event
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/lib/openapi/Namf_EventExposure/logger/logger.go:62 free5gc/lib/openapi/Namf_EventExposure/logger.SetReportCaller() set report call : true                        Namf=Event
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/lib/openapi/Nsmf_PDUSession/logger/logger.go:57 free5gc/lib/openapi/Nsmf_PDUSession/logger.SetLogLevel() set log level : info                          Nsmf=PDUSession
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/lib/openapi/Nsmf_PDUSession/logger/logger.go:62 free5gc/lib/openapi/Nsmf_PDUSession/logger.SetReportCaller() set report call : true                        Nsmf=PDUSession
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/lib/openapi/Nudr_DataRepository/logger/logger.go:57 free5gc/lib/openapi/Nudr_DataRepository/logger.SetLogLevel() set log level : info                          Nudr=datarepository
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/lib/openapi/Nudr_DataRepository/logger/logger.go:62 free5gc/lib/openapi/Nudr_DataRepository/logger.SetReportCaller() set report call : true                        Nudr=datarepository
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/lib/openapi/logger/logger.go:57 free5gc/lib/openapi/logger.SetLogLevel() set log level : info                          OpenApi=OpenApi
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/lib/openapi/logger/logger.go:62 free5gc/lib/openapi/logger.SetReportCaller() set report call : true                        OpenApi=OpenApi
INFO[2020-09-18T06:04:10Z] set log level : info                          aper=aper
INFO[2020-09-18T06:04:10Z] set report call : true                        aper=aper
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/lib/CommonConsumerTestData/logger/logger.go:60 free5gc/lib/CommonConsumerTestData/logger.SetLogLevel() set log level : info                          CommonTest=comm
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/lib/CommonConsumerTestData/logger/logger.go:65 free5gc/lib/CommonConsumerTestData/logger.SetReportCaller() set report call : true                        CommonTest=comm
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/src/smf/factory/factory.go:36 free5gc/src/smf/factory.InitConfigFactory() Successfully initialize configuration ../config/smfcfg.conf  SMF=init
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/src/smf/factory/factory.go:48 free5gc/src/smf/factory.InitRoutingConfigFactory() Successfully initialize configuration ../config/uerouting.yaml  SMF=init
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/src/smf/service/init.go:101 free5gc/src/smf/service.(*SMF).Initialize() Log level is set to [info] level              SMF=init
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/src/smf/context/smf_context.go:78 free5gc/src/smf/context.InitSmfContext() smfconfig Info: Version[1.0.0] Description[AMF initial local configuration]  SMF=Context
INFO[2020-09-18T06:04:10Z]/go/src/free5gc/src/smf/service/init.go:130 free5gc/src/smf/service.(*SMF).Start() Server started                                SMF=init
INFO[2020-09-18T06:04:14Z]/go/src/free5gc/src/smf/consumer/nnrf.go:61 free5gc/src/smf/consumer.SendNFRegistration() SMF Registration to NRF {1e2ce6e9-5798-4205-a6bd-43990eb4a573 SMF REGISTERED 0 0xc000106400 <nil> [] []   [smf] [] <nil> [] [] <nil> 0 0 0  <nil> <nil> <nil> <nil> 0xc00010e340 <nil> <nil> <nil> <nil> <nil> map[] <nil> false 0xc0001062e0 false false []}  SMF=init
INFO[2020-09-18T06:04:14Z]/go/src/free5gc/src/smf/pfcp/udp/udp.go:35 free5gc/src/smf/pfcp/udp.Run() Listen on 10.100.200.12:8805                  SMF=pfcp
INFO[2020-09-18T06:04:14Z]/go/src/free5gc/src/smf/service/init.go:169 free5gc/src/smf/service.(*SMF).Start() Send PFCP Association Request to UPF[117.112.102.49:8805]  SMF=app
INFO[2020-09-18T06:04:14Z]/go/src/free5gc/src/smf/service/init.go:169 free5gc/src/smf/service.(*SMF).Start() Send PFCP Association Request to UPF[117.112.102.50:8805]  SMF=app
INFO[2020-09-18T06:04:14Z]/go/src/free5gc/src/smf/service/init.go:169 free5gc/src/smf/service.(*SMF).Start() Send PFCP Association Request to UPF[117.112.102.98:8805]  SMF=app
INFO[2020-09-18T06:04:23Z]/go/src/free5gc/lib/pfcp/pfcpUdp/udp.go:143 free5gc/lib/pfcp/pfcpUdp.(*PfcpServer).RemoveTransaction() Remove Request Transaction [1]                PFCP=PFCP
INFO[2020-09-18T06:04:23Z]/go/src/free5gc/lib/pfcp/pfcpUdp/udp.go:143 free5gc/lib/pfcp/pfcpUdp.(*PfcpServer).RemoveTransaction() Remove Request Transaction [2]                PFCP=PFCP
INFO[2020-09-18T06:04:23Z]/go/src/free5gc/lib/pfcp/pfcpUdp/udp.go:143 free5gc/lib/pfcp/pfcpUdp.(*PfcpServer).RemoveTransaction() Remove Request Transaction [3]                PFCP=PFCP
```

The node_ids are configured as "upf1", "upf2", and "upfb".
In the logs, PFCP Association Requests are sent to 117.112.102.49, 117.112.102.50, and 117.112.102.98.
Those are not valid addresses for UPFs.
("upf1" in ascii code representation is "117 112 102 49")

### Describe the solution you'd like
Resolve hostname when loading the node_ids from smfcfg.conf.

### Describe alternatives/workarounds you've considered
Resolve hostname before sending PFCP Association Requests.

### Additional context
None.

### Would you be willing to resolve this issue by submitting a Pull Request?

- [x] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by sh-sekigawa** [comment]
I've write a patch for this issue.
https://github.com/sh-sekigawa/smf/commit/8c8b76c2e7287791b14dd89a262de90ecccc77b7

Should I create PR on https://github.com/free5gc/smf?

**Comment by free5gc-org** [comment]
Dear @sh-sekigawa ,

Many thanks for your contribution. However, we'll have a latest release in this or next month.
We'll recommend you to create PR after our next release.

Many thanks for your help again! :+1:

**Comment by free5gc-org** [comment]
Hi @sh-sekigawa ,

We have release our new version of v3.0.4.
It's very thanks if you can give us a PR for this issue.

**Comment by sh-sekigawa** [comment]
Thanks for letting me know.
I'll create PR after testing my patch on your new release.

**Comment by sh-sekigawa** [comment]
I've confirmed that the issue has been solved in v3.0.4.
